tools/operator/scripts/validate.sh

```diff
@@ -0,0 +1,146 @@
+do
+  CHANNELS+=("testorgschannel$i")
+done
+
+ORDERER_ORG=${ORDERER_ORGS[0]}
+export ORDERER_NAME=orderer0-$ORDERER_ORG
+CURRENT_DIR=$(cd `dirname $0` && pwd)
+export FABRIC_CFG_PATH=$CURRENT_DIR/../../../fabric/.build/config/
+
+
+exportOrdererConfig(){
+  export CORE_PEER_MSPCONFIGPATH="$ARTIFACTS_LOCATION/crypto-config/$4Organizations/$1/users/Admin@$1/msp"
+  export CORE_PEER_TLS_ROOTCERT_FILE="$ARTIFACTS_LOCATION/crypto-config/$4Organizations/$1/$4s/$2.$1/tls/ca.crt"
+  export ORDERER_NAME=$2
+  NODEPORT=$(kubectl --kubeconfig=$KUBECONFIG get -o jsonpath="{.spec.ports[0].nodePort}" services $2)
+  NODEIP=$(kubectl --kubeconfig=$KUBECONFIG get nodes -o jsonpath='{ $.items[*].status.addresses[?(@.type=="ExternalIP")].address }' | cut -d' ' -f1)
+  export CORE_PEER_ADDRESS=$NODEIP:$NODEPORT
+  CHANNEL_NAME=$3
+  if [[ "$VALIDATE_BLOCK" == "true" ]]; then
+    if [[ "$4" == "peer" ]]; then
+      peerBlockValidation
+    else
+      ordererBlockValidation
+    fi
+  fi
+  if [[ -n "$CONSENSUS_CHECK" && "$CONSENSUS_CHECK" == "true" ]]; then
+    peer channel fetch config config_block.pb -o $CORE_PEER_ADDRESS -c $CHANNEL_NAME --tls --cafile $CORE_PEER_TLS_ROOTCERT_FILE --ordererTLSHostnameOverride $ORDERER_NAME
+    configtxlator proto_decode --input config_block.pb --type common.Block | jq .data.data[0].payload.data.config > config.json 
+    validateConsensus $CONSENSUS_TYPE $CONSENSUS_STATE
+  fi
+}
+
+validateConsensus(){
+  consensusType=$(jq -r '.channel_group.groups.Orderer.values.ConsensusType.value.type' config.json)
+  consensusState=$(jq -r '.channel_group.groups.Orderer.values.ConsensusType.value.state' config.json)
+  if [[ "${consensusType}" == "$1" && "${consensusState}" == "$2" ]]; then
+    echo "Consensus type:$consensusType and state:$consensusState"
+  else
+    echo "$CHANNEL_NAME failed to migrate to etcdraft"
+    exit 1
+  fi
+}
+
+ordererBlockValidation(){
+  peer channel fetch newest config_block.pb -o $CORE_PEER_ADDRESS -c $CHANNEL_NAME --tls --cafile $CORE_PEER_TLS_ROOTCERT_FILE --ordererTLSHostnameOverride $ORDERER_NAME > block.txt 2>&1
+  blockNum=`cat block.txt | grep "Received block" | awk '{print $11}'`
+  validateBlock $blockNum
+}
+
+peerBlockValidation(){
+    export CORE_PEER_TLS_ENABLED=true
+    peer channel getinfo -o $CORE_PEER_ADDRESS -c $CHANNEL_NAME --tls > block.txt 2>&1
+    cat block.txt | grep '"height":' | awk '{print $3}' > block.json
+    blockHeight=$(jq -r '.height' block.json)
+    validateBlock $blockHeight
+    rm -rf block.json
+}
+
+validateBlock(){
+  echo "--------------------------------------------------------------------------"
+  echo "Block validation for $ORDERER_NAME in $CHANNEL_NAME"
+  echo "--------------------------------------------------------------------------"
+  blockNum=$1
+  if [ $FLAG -gt 0 ]; then
+    i=0
+    while [ $i -lt 3 ];
+    do
+      if [ $blockNum != $LATEST_BLOCK ]; then
+        i=3
+      elif [ $blockNum != $LATEST_BLOCK && $i -lt 3 ]; then
```

Comment:
Maybe we could change from != to < .  That way, we would be verifying that this blocknum is AT LEAST caught up. That might help if traffic is running.

Reply:
okay. Will change this

Reply:
That seems more robust. THanks.

Reply:
Done